pkg/cmd/cli/cmd/env.go

```diff
@@ -232,6 +234,21 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	if !one && len(resourceVersion) > 0 {
 		return cmdutil.UsageError(cmd, "--resource-version may only be used with a single resource")
 	}
+	// Keep a copy of the original objects prior to updating their environment.
+	// Used in constructing the patch(es) that will be applied in the server.
+	oldObjects, err := resource.AsVersionedObjects(infos, outputVersion)
```

Comment:
Not necessarily - clientConfig.Version becomes the default if no output
version is provided.

On Nov 19, 2015, at 11:50 AM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/cmd/cli/cmd/env.go
<https://github.com/openshift/origin/pull/5819#discussion_r45364573>:

> @@ -232,6 +234,21 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
>  	if !one && len(resourceVersion) > 0 {
>  		return cmdutil.UsageError(cmd, "--resource-version may only be used with a single resource")
>  	}
> +	// Keep a copy of the original objects prior to updating their environment.
> +	// Used in constructing the patch(es) that will be applied in the server.
> +	oldObjects, err := resource.AsVersionedObjects(infos, outputVersion)

outputVersion := cmdutil.OutputVersion(cmd, clientConfig.Version) ensures
we're working with the same version we're going to submit to the server,
right?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5819/files#r45364573>.